runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -259,7 +259,7 @@ public void onChatMessage(ChatMessage event)
 	@Subscribe
 	public void onExperienceChanged(ExperienceChanged event)
 	{
-		if (!config.enabled() || event.getSkill() != SLAYER)
+		if (!config.enabled() || event.getSkill() != SLAYER || client.getSkillExperience(SLAYER) <= cachedXp)
```

Comment:
Save the client.getSkillExperience(SLAYER) to variable, as you are calling the same method 3 times in span of like 10 lines of code.